src/Console/Command/DescribeCommand.php

```diff
@@ -0,0 +1,227 @@
+    {
+        $fixerFactory = new FixerFactory();
+        $fixers = array();
+
+        foreach ($fixerFactory->registerBuiltInFixers()->getFixers() as $fixer) {
+            $fixers[$fixer->getName()] = $fixer;
+        }
+
+        if (!isset($fixers[$name])) {
+            throw new \InvalidArgumentException(sprintf('Rule "%s" does not exist.', $name));
+        }
+
+        $fixer = $fixers[$name];
+        $description = $fixer instanceof FixerDescriptionAwareInterface ? $fixer->getFixerDescription() : null;
+
+        $output->writeln(sprintf('<info>Description of %s rule.</info>', $name));
+        $output->writeln($fixer->getDescription());
+        $output->writeln('');
+
+        if ($fixer->isRisky()) {
+            $output->writeln('<error>Fixer applying this rule is risky.</error>');
+
+            if ($description && $description->getRiskyDescription()) {
+                $output->writeln($description->getRiskyDescription());
+            }
+
+            $output->writeln('');
+        }
+
+        if ($this->isFixerConfigurable($fixer)) {
+            $output->writeln('<comment>Fixer is configurable.</comment>');
+
+            if ($description && $description->getConfigurationDescription()) {
+                $output->writeln($description->getConfigurationDescription());
+            }
+
+            if ($description && $description->getDefaultConfiguration()) {
+                $output->writeln(sprintf('Default configuration: <comment>%s</comment>.', $this->arrayToText($description->getDefaultConfiguration())));
+            }
+
+            $output->writeln('');
+        }
+
+        if ($description && $description->getCodeSamples()) {
+            $output->writeln('Fixing examples:');
+            $differ = new SebastianBergmannDiffer();
+
+            foreach ($description->getCodeSamples() as $index => $codeSample) {
+                $old = $codeSample[0];
+                $tokens = Tokens::fromCode($old);
+                $fixer->configure($codeSample[1]);
+                $fixer->fix(new StdinFileInfo(), $tokens);
+                $new = $tokens->generateCode();
+                $diff = $differ->diff($old, $new);
+
+                if (null === $codeSample[1]) {
+                    $output->writeln(sprintf(' * Example #%d.', $index + 1));
+                } else {
+                    $output->writeln(sprintf(' * Example #%d. Fixing with configuration: <comment>%s</comment>.', $index + 1, $this->arrayToText($codeSample[1])));
+                }
+                $output->writeln($this->prepareDiff(true, $diff));
+                $output->writeln('');
+            }
+        }
+
+        if (!$description) {
+            $output->writeln(sprintf('<question>This rule is not yet described, do you want to help us and describe it?</question>'));
+            $output->writeln('Contribute at <comment>https://github.com/FriendsOfPHP/PHP-CS-Fixer</comment>');
+            $output->writeln('');
```

Comment:
One challenge is to provide nice mechanism for describing rules / sets with nice console output.
This PR tries to do that.

Second challenge is to provide that descriptions for all rules. For ~130 it's not trivial, and I think we don't need to do it for all fixers in advance, before releaseing description command.